src/DistIL/AsmIO/ResolvingUtils.cs

```diff
@@ -11,19 +11,46 @@ public static class ResolvingUtils
     /// <returns></returns>
     public static MethodDesc? FindMethod(this ModuleResolver resolver, string selector)
     {
-        if (resolver.FunctionCache.TryGetValue(selector, out MethodDesc? cachedMethod)) {
+        var convertedSelector = GetSelector(resolver, selector);
+        if (resolver.FunctionCache.TryGetValue(convertedSelector, out MethodDesc? cachedMethod)) {
             return cachedMethod;
         }
 
+        if (convertedSelector.Type == null) {
+            return null;
+        }
+
+        return ResolveMethod(resolver, convertedSelector, convertedSelector);
+    }
+
+    /// <summary>
+    /// Finds a method based on a selector and arguments
+    /// </summary>
+    /// <param name="resolver"></param>
+    /// <param name="selector">Specifies which method to select</param>
+    /// <example>FindMethod("System.Text.StringBuilder::AppendLine", [ConstString.Create("Hello World")])</example>
+    /// <returns></returns>
+    public static MethodDesc? FindMethod(this ModuleResolver resolver, string selector, IEnumerable<Value> values)
+    {
         var convertedSelector = GetSelector(resolver, selector);
+        convertedSelector.ParameterTypes.AddRange(values.Select(value => value.ResultType));
+
+        if (resolver.FunctionCache.TryGetValue(convertedSelector, out MethodDesc? cachedMethod)) {
+            return cachedMethod;
+        }
```

Comment:
I think caching won't work with this change because lists are compared by reference by default, Selector needs to override `Equals` and `GetHashCode` and compare with `SequenceEqual`